langchain4j/src/main/java/dev/langchain4j/service/ServiceOutputParser.java

```diff
@@ -123,7 +123,9 @@ private static String jsonStructure(Class<?> structured) {
         jsonSchema.append("{\n");
         for (Field field : structured.getDeclaredFields()) {
             String name = field.getName();
-            if (name.equals("__$hits$__")) {
+            if (name.equals("__$hits$__")
+                    || java.lang.reflect.Modifier.isStatic(field.getModifiers())
+                    || java.lang.reflect.Modifier.isFinal(field.getModifiers())) {
```

Comment:
Unfortunately this line is severely breaking. Consider `ServiceOutputParserTest.Person`: By simply making is class immutable (and adding a constructor so JSON deserialization works), `outputFormatInstructions` will not return `{\n}`.
Furthermore, the same problem occurs for records.

I am not sure what the motivation for this line was, but I propose at the very least, it should be removed

Reply:
@geoand thanks for the heads up! Does it break Quarkus extension?

Reply:
@tenpigs267 BTW it also does not work if you have a recursion. So if I define it like this:

```java
class Person {

            private String firstName;
            private String lastName;
            private LocalDate birthDate;
            private List<Person> parents;
```

It will go into endless recursion.

Reply:
> @geoand thanks for the heads up! Does it break Quarkus extension?

It breaks some of our tests that use Java Records

Reply:
thanks @langchain4j for #620 and sry for that.
For the recursive thing I'am going to add a PR with the code going through all involved classes and generating a list a class (and test it with openAI)